### ISSUE TITLE
Apache and DNS on IPv6-only systems got issues

### ISSUE THREAD

**Issue by iliajie** [virtualmin/virtualmin-gpl#980]
Hello Jamie!

Since [this PR](https://github.com/virtualmin/virtualmin-gpl/pull/974) has been merged, I’d like to create a ticket to keep track of other issues that still haven’t been addressed.

In short, Apache records on IPv6-only systems are still being created as:

```
<VirtualHost :80 [fdb2:2c26:f4e4:0:21c:2222:1111:0000]:80>
```

And DNS zone records are being created as:

```
domain.tld.      IN      A       fdb2:2c26:f4e4:0:21c:2222:1111:0000
```

**Comment by jcameron** [comment]
Wow that DNS record seems wrong! How did you trigger this behavior exactly? Is it now possible to create an IPv6-only domain from the CLI that has this problem?

**Comment by iliajie** [comment]
Previously, it wasn’t even possible to create a new virtual server on IPv6-only system. Now, after the previous PR was merged, it is possible. However, despite running the config check and the removal of cached IPv4 record from Virtualmin config, i.e.:

<img width="314" alt="image" src="https://github.com/user-attachments/assets/16d536df-c050-40ce-a697-01afd0f10dd2">


Using the "Create Virtual Server" page still generates broken records for Apache and BIND on all IPv6-only systems:

<img width="891" alt="image" src="https://github.com/user-attachments/assets/89970ce1-0fdb-424b-8b8a-2f1e37c52bd8">

Creates broken Apache and BIND configs, e.g.:

<img width="353" alt="image" src="https://github.com/user-attachments/assets/5e83fddb-5e8d-478f-9a7b-092661dc9ed7">

<img width="743" alt="image" src="https://github.com/user-attachments/assets/e0529b8d-6abb-45d2-97e5-29bb0a11238c">

As you can see, the public IPv6 address isn’t being used correctly. Interestingly, DNS `A` records receive the public IPv6 address, while `AAAA` records get the local IPv6 address. Apache doesn’t get anything for IPv4 and just prints the port, like `:80`.

**Comment by jcameron** [comment]
Ok that Apache config is definitely a bug! This should fix it though : https://github.com/virtualmin/virtualmin-gpl/commit/f2b9e30ecbb73834a19093298d4889dbb926da67

**Comment by iliajie** [comment]
> Ok that Apache config is definitely a bug! This should fix it though : [f2b9e30](https://github.com/virtualmin/virtualmin-gpl/commit/f2b9e30ecbb73834a19093298d4889dbb926da67)

Giving it a quick look, it seems to resolve the issue. Do we need a similar patch for Nginx?

> As for the DNS issue, what does the dns_ip= line in the domain's config file under /etc/webmin/virtual-server/domains contain?

Check this out but note that `173376457750228` is the new domain, created after the system was switched to IPv6-only network.

<img width="766" alt="image" src="https://github.com/user-attachments/assets/c12bcecf-c6c4-416f-aa98-941e9c6c75e8">

**Comment by jcameron** [comment]
Ok that's the problem, if `dns_ip` is set to an IPv6 address it will cause this kind of problem. It should be empty.

**Comment by jcameron** [comment]
Yes there will be a LOT of work to make IPv6-only mode work! But for starters, the `ip` field should certainly not be set, and likely `dns_ip` will not be set unless there is an odd case where the system only has a v6 address but is behind some kind of NAT that gives it an external v4 address.

**Comment by iliajie** [comment]
Well, okay! With all the recent changes checked into master, `$ip` shouldn’t be set anymore already. However, as you can see in the screenshot above, `dns_ip`, even if empty, is still being added to DNS.

Let’s leave it for now—can you add this to your distant to-do list?

By the way, SSL certificate sharing also isn’t working correctly in IPv6-only mode...

**Comment by jcameron** [comment]
Sure, I will update this ticket with progress on IPv6-only support

**Comment by iliaross** [comment]
Do you think it’ll be possible to have proper support for IPv6-only systems in Virtualmin 8?

**Comment by iliaross** [comment]
> Unlikely, there are a _lot_ of code changes needed to remove the assumptions about IPv4 being available.

Yeah, alright... And, I still think we have some time before this becomes necessary. For example, my home internet only got IPv6 support about six months ago, and I know many still don’t have it.

So, let’s keep this open with the plan to make it work eventually...

Thanks!

**Comment by jcameron** [comment]
Yeah I think it's super rare that anyone would want to host an IPv6 only website on the public internet. There may be legitimate intranet use-cases though.

**Comment by iliaross** [comment]
Reopened, as this was closed by an automated cleanup run.

⚠️ If the issue is resolved, please take a moment to close it to help keep the project organized.

**Comment by jcameron** [comment]
FYI, the next Virtualmin release will support IPv6-only domains for all features!
